paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/kafka/KafkaLogStoreRegister.java

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.kafka;
+
+import org.apache.paimon.catalog.Identifier;
+import org.apache.paimon.flink.log.LogStoreRegister;
+import org.apache.paimon.flink.log.LogStoreTableFactory;
+import org.apache.paimon.utils.Preconditions;
+
+import org.apache.paimon.shade.guava30.com.google.common.collect.ImmutableMap;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.paimon.flink.FlinkCatalogOptions.REGISTER_TIMEOUT;
+import static org.apache.paimon.flink.FlinkConnectorOptions.LOG_SYSTEM_PARTITIONS;
+import static org.apache.paimon.flink.FlinkConnectorOptions.LOG_SYSTEM_REPLICATION;
+import static org.apache.paimon.flink.kafka.KafkaLogOptions.BOOTSTRAP_SERVERS;
+import static org.apache.paimon.flink.kafka.KafkaLogOptions.TOPIC;
+
+/** KafkaLogStoreRegister is used to register/unregister topics in Kafka for paimon table. */
+public class KafkaLogStoreRegister implements LogStoreRegister {
+
+    private final String bootstrapServers;
+
+    private final String topic;
+
+    private final int partition;
+
+    private final int replicationFactor;
+
+    private final Duration timeout;
+
+    private final Properties properties;
+
+    private final Identifier identifier;
+
+    public KafkaLogStoreRegister(LogStoreTableFactory.RegisterContext context) {
+        this.bootstrapServers = context.getOptions().get(BOOTSTRAP_SERVERS);
+        this.identifier = context.getIdentifier();
+        this.topic =
+                context.getOptions().get(TOPIC) != null
+                        ? context.getOptions().get(TOPIC)
+                        : String.format(
+                                "%s_%s_%s",
+                                this.identifier.getDatabaseName(),
+                                this.identifier.getObjectName(),
+                                UUID.randomUUID().toString().replace("-", ""));
+
+        Preconditions.checkArgument(this.bootstrapServers != null);
+        Preconditions.checkArgument(this.topic != null);
+        Preconditions.checkArgument(this.identifier != null);
+
+        // handle the type information missing when Map is converted to Options
+        if (context.getOptions().get(REGISTER_TIMEOUT.key()) == null) {
+            this.timeout = REGISTER_TIMEOUT.defaultValue();
+        } else {
+            this.timeout = Duration.parse(context.getOptions().get(REGISTER_TIMEOUT.key()));
+        }
+        this.partition = context.getOptions().get(LOG_SYSTEM_PARTITIONS);
```

Comment:
How about use bucket count when user does not configure partitions for log system?